File: java/src/test/java/problems101_200/BalancedBinaryTreeTest.java
package problems101_200;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class BalancedBinaryTreeTest extends SampledTest {
    @Test
    public void test() {
        for (BalancedBinaryTree solution : BalancedBinaryTree.values()) {
            for (BalancedBinaryTreeSample sample : getSamples(BalancedBinaryTreeSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isBalanced(TreeNode.of(sample.input.root)))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/BalancedBinaryTreeTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class BalancedBinaryTreeTest : SampledTest() {
    @Test
    fun test() =
        BalancedBinaryTree.entries.forEach { solution ->
            getSamples<Array<BalancedBinaryTreeSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.isBalanced(TreeNode.of(*input.root)))
                        .isEqualTo(output)
                }
        }
}
